sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java

```diff
@@ -0,0 +1,306 @@
+                + "YE4yV3E4YmrJjggH1FOVa1wgCeGQKBgQCbCKeM4EahWIyTBiZsTQ/l5hwhjPCrxncbyA2EZWNg3Ri6nuMIQzoBrCoX9L8t7e0"
+                + "CKWAN0oM2Cn1VIJhsiE75dzN3vvGBcNZ9y+BwbwxDIAhrztyKKJS0h9YmAUVr+w5WsUPyMUPQ0/1wdTdxvKqQpriddrvyKRSJ"
+                + "M9fb29+cwQ==";
+        }
+        return signingKey;
+    }
+
+    String getPolicySigningCertificate0() {
+        String certificate = Configuration.getGlobalConfiguration().get("policySigningCertificate0");
+        if (certificate == null) {
+            certificate = "MIIC1jCCAb6gAwIBAgIIAxfcH6Co5DowDQYJKoZIhvcNAQELBQAwIjEgMB4GA1UEAxMXQXR0ZXN0YXRpb25DZXJ"
+                + "0aWZpY2F0ZTAwHhcNMjEwMTE5MjAxMjU2WhcNMjIwMTE5MjAxMjU2WjAiMSAwHgYDVQQDExdBdHRlc3RhdGlvbkNlcnRpZmlj"
+                + "YXRlMDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAOOpb5GvUCuOYiB4ZazIePtSazdXGDyjtFlr4ulo1VY1Ai91X"
+                + "IcWIPELCV1OfQiIoJlj096u3cirP1GvCKgb4FTNHHi7omDaQvYRmuZZ6KXrqNi5Iu/jKjGgjwYt+FYV/9eqYCWdyS0RjMbKw7"
+                + "sZUvBxTDeTqQunwbjPZ1y4JbxXx6xwcZJHfwD6g7aHslsblHh4zM1mhiuoIMpNUeeThLwQTD6oGSmIt+hqRbfvd3Ljr/v7W3m"
+                + "SKvw5X9L85PNHaDIUd4vHSDiytZUoXyhtbC8RKGzxgZCz6gFwM5JF6QhYE/A84HFH7JZ3FKk1UJBoTjcv63BshT7Pt3fYMZqV"
+                + "SzkCAwEAAaMQMA4wDAYDVR0TBAUwAwEB/zANBgkqhkiG9w0BAQsFAAOCAQEAC+YKGp6foV94lYQB/yNQ53Bk+YeR4dgAkR99U"
+                + "t1VvvXKyWnka/X8QRjKoaPULDZH4+9ZXg6lSfhuEGTSPUzO294mlNbF6n6cuuuawe3OeuZUO53b4xYXPRv898FBRxdD+FCW/f"
+                + "A5HLBrGItfk31+aNUFryCd5RrJfJU8Rurm+7uGPtS16Ft0P7xSnL0C7nfHNVuEKFV0ZbzgzXlzkKQT4d3fYpvOxzYoXImxzwz"
+                + "W/jzZjN3aKbOlmY2LyW8J5BKKgA3C4FRWwCTmgqYp2vQhsw1HgCeBjmBN5/imnk2lsgjrvvSdlkXOnNf5atibuguYzdakz99b"
+                + "wwWWsd5HddtcyA==";
+        }
+        return certificate;
+    }
+
+    private static String getLocationShortName()
+    {
+        String shortName = Configuration.getGlobalConfiguration().get("locationShortName");
+        if (shortName == null) {
+            shortName = "wus";
+        }
+        return shortName;
+    }
+
+    private static String getIsolatedUrl()
+    {
+        String url = Configuration.getGlobalConfiguration().get("ATTESTATION_ISOLATED_URL");
+        if (url == null) {
+            url = "https://attestation_isolated_url";
+        }
+        return url;
+    }
+
+    private static String getAadUrl()
+    {
+        String url = Configuration.getGlobalConfiguration().get("ATTESTATION_AAD_URL");
+        if (url == null) {
+            url = "https://attestation_aad_url";
+        }
+        return url;
+    }
+
+    static Stream<Arguments> getAttestationClients() {
+        // when this issues is closed, the newer version of junit will have better support for
+        // cartesian product of arguments - https://github.com/junit-team/junit5/issues/1427
+
+        List<Arguments> argumentsList = new ArrayList<>();
+
+        String regionShortName = getLocationShortName();
+        getHttpClients().forEach(httpClient -> Stream.of(
```

Comment:
Instead of allocating an ArrayList, why not use stream operators to transform the clients? Same with the one below
```java
final String regionShortName = getLocationShortName();
return getHttpClients().flatMap(httpClient -> {
    return Stream.of(
        Arguments.of(httpClient, "https://shared" + regionShortName + "." + regionShortName + ".test.attest.azure.net"),
        Arguments.of(httpClient, getIsolatedUrl()),
        Arguments.of(httpClient, getAadUrl())
    );
```